pkg/mdm/mdmtest/apple.go

```diff
@@ -199,6 +228,119 @@ func (c *TestAppleMDMClient) fetchEnrollmentProfileFromDEPURL() error {
 	)
 }
 
+func (c *TestAppleMDMClient) fetchEnrollmentProfileFromOTAURL() error {
+	rawDeviceInfo := []byte(fmt.Sprintf(`<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>PRODUCT</key>
+	<string>%s</string>
+	<key>SERIAL</key>
+	<string>%s</string>
+	<key>UDID</key>
+	<string>%s</string>
+	<key>VERSION</key>
+	<string>22A5316k</string>
+</dict>
+</plist>`, c.Model, c.SerialNumber, c.UUID))
+
+	do := func(cert *x509.Certificate, key *rsa.PrivateKey) ([]byte, error) {
+		signedData, err := pkcs7.NewSignedData(rawDeviceInfo)
+		if err != nil {
+			return nil, fmt.Errorf("create signed data: %w", err)
+		}
+		err = signedData.AddSigner(cert, key, pkcs7.SignerInfoConfig{})
+		if err != nil {
+			return nil, fmt.Errorf("add signer: %w", err)
+		}
+		sig, err := signedData.Finish()
+		if err != nil {
+			return nil, fmt.Errorf("finish signing: %w", err)
+		}
+
+		request, err := http.NewRequest(
+			"POST",
+			c.fleetServerURL+"/api/latest/fleet/ota_enrollment?enroll_secret="+c.otaEnrollSecret,
+			bytes.NewReader(sig),
+		)
+		if err != nil {
+			return nil, fmt.Errorf("create request: %w", err)
+		}
+		// #nosec (this client is used for testing only)
+		cc := fleethttp.NewClient(fleethttp.WithTLSClientConfig(&tls.Config{
+			InsecureSkipVerify: true,
+		}))
+		response, err := cc.Do(request)
+		if err != nil {
+			return nil, fmt.Errorf("send request: %w", err)
+		}
```

Comment:
Missing `defer response.Body.Close()`. Might want to check if other enrollment types also leak the response body, this could potentially amount to some device-side issues in load tests (although we seem to create a new client on every enroll method, so probably not, but still better to make sure we do close it).

Reply:
awesome catch, thanks! It was indeed missing in a few places here. Added!